src/adapter/src/coord/caught_up.rs

```diff
@@ -385,4 +419,139 @@ impl Coordinator {
             ctx.trigger.fire();
         }
     }
+
+    /// Analyzes replica status history to detect replicas that are
+    /// crash-looping or OOM-looping.
+    ///
+    /// A replica is considered problematic if it has multiple failures in a
+    /// short time window. This includes crashes, OOM kills, or other non-ready
+    /// states that repeat frequently.
```

Comment:
I think the comment is out of sync with `is_problematic_status`.

Reply:
will fix!